### ISSUE TITLE
Inconsistency in Transaction Log and Smesher Log

### ISSUE THREAD

**Issue by whoabuddy** [spacemeshos/smapp#535]
**Describe the bug**

While sending some very merry unbirthday coins around, I noticed an inconsistency between the transaction log activity stats and the smesher log. The transaction log does not include my true smeshing rewards, and the totals computed are incorrect.

**To Reproduce**

1. Install the smapp
2. Run the smapp for at least 48-72 hours (to get some rewards)
3. Check the transaction log against the smesher / smesher log

**Expected behavior**

I expected the transaction log to reflect the activities recorded in the smesher log, including the rewards received between `Jun 9 2020, 10:35:52 AM` and `Jun 11 2020, 9:05:52 AM`.

I also expected the activity, currently set to monthly, to reflect the correct stats since smeshing occurred in June (so far).

Same issue for **Activity** and **Total**:

- Smeshed: 0.943 SMH *(should be 258.583 SMH)*
- Sent: 7 SMH *(correct)*
- Received: 300 SMD *(correct)*

**Screenshots**

![20200611 inconsistency in transaction log and activity](https://user-images.githubusercontent.com/9038904/84412572-4ce16100-abc4-11ea-9a56-6b15ab1063da.png)

**Desktop (please complete the following information):**

- OS: Ubuntu Desktop 20.04
- Smapp: Release 0.0.13

**Additional context**

I am not sure what the value of **Monthly coins:** should be, it currently shows 7.943 SMH which would be total smeshed + total sent, but it does not include the 300 SMD received (or rounds it away, based on SMH vs SMD size).

If the intent is to show total amount monthly ins + monthly outs, then I think it needs a new name.

If the intent is to show the total money generated in the activity period, then it should subtract the amount sent.

**Comment by whoabuddy** [comment]
Interestingly enough, after filing this issue I noticed that the smeshing rewards are now showing in the transaction log, starting after my last transfer at `June 11 2020, 09:20:52 AM`, so maybe it got "stuck" for some reason?

![VirtualBox_SmeshLikeAMofo_11_06_2020_13_27_48](https://user-images.githubusercontent.com/9038904/84435975-8aa3b100-abe7-11ea-8d70-27961541bafa.png)

**Comment by IlyaVi** [comment]
@whoabuddy please attach link to the spacemesh-log.txt file

**Comment by whoabuddy** [comment]
@IlyaVi sorry for the delay, had some issues creating a shared folder from the VM to transfer, but I did make a copy of the log and compressed it. I will try to get it uploaded tomorrow! (And install the latest version of the smapp)

**Comment by whoabuddy** [comment]
@IlyaVi Here you go! https://envl.app/whoabuddy/ydTGfT!8dUHP04TCK9nzSTl

**Comment by IlyaVi** [comment]
@antonlerner @noamnelke please look at the log file from smesher point of view and I will check app's requests

**Comment by noamnelke** [comment]
@IlyaVi 
Can you summarize the issue from your point of view? When I look at this I don't understand which api endpoint gets called to produce the smeshing rewards in the "transaction log" (only showing the last one for a while) vs. "smesher log" (showing everything correctly).

I also don't know how "monthly coins" and "smeshed" are calculated.

It would be much easier for me to investigate if you told me something like "X api was called and returned everything correctly, while Y api was called around the same time and only returned the last smeshing reward, later Y api was called again and returned everything correctly".

Regarding the "monthly coins" issue, I have no idea what that means, how it's calculated or why it's useful. Maybe @avive can help figure it out. 

@avive, while you're here: I also think that SMH vs. SMD is **_very_** confusing, and I'm glad that @whoabuddy figured out that the SMDs are being rounded away, because I may have taken forever to catch that... I strongly believe (I've said it before) that we should refrain from using more than one unit of account and I also think we should consistently show the same number of decimal points on all numbers. If this is a problem due to very small fees then we should find a UI solution (for example, round the fees away and show the full number on hover) but showing different coin ticker symbols for fees/small amounts vs. mined coins is a terrible solution from a user perspective, imo.

**Comment by avive** [comment]
#393

**Comment by menzels** [comment]
I just want to add one datapoint here because i have the same right now.

1. Setup wallet start smeshing
2. after a day i got quite a few smeshing rewards and they are displayed nicely in the transaction log
3. i send coins the first time
4. now only the first smeshing reward is displayed in the transaction log
5. the main wallet balance does still include the smeshing reward.

**Comment by avive** [comment]
@IlyaVi - in both views - smesher screen and wallet, rewards should be displayed sorted by timestamp descending so the most recent ones are displayed first.
